storage/src/main/java/tech/pegasys/artemis/storage/client/RecentChainData.java

```diff
@@ -130,9 +134,22 @@ public Store getStore() {
    * @param slot
    */
   public void updateBestBlock(Bytes32 root, UnsignedLong slot) {
+    final Optional<Bytes32> originalBestRoot = bestBlockRoot;
+    final UnsignedLong originalBestSlot = bestSlot;
     this.bestBlockRoot = Optional.of(root);
     this.bestSlot = slot;
     bestBlockInitialized.complete(null);
+
+    if (originalBestRoot.map(original -> isReorg(original, originalBestSlot)).orElse(false)) {
+      reorgEventChannel.reorgOccurred();
+    }
+  }
+
+  private boolean isReorg(final Bytes32 originalBestRoot, final UnsignedLong originalBestSlot) {
+    // Get the block root in effect at the old best slot on the new chain
```

Comment:
nit: A bit of explanation on how getBlockRootBySlot might return a different root at that slot would be nice, since the updating of the canonical chain with the switch to a new bestBlockRoot is a bit implicit.
```suggestion
    // Get the block root in effect at the old best slot on the new chain, which might be different from originalBestRoot due to getBlockRootBySlot potentially using a new chain as the correct fork

Reply:
Renamed the method and improved the comment to make it clearer.